waves-ext/src/main/scala/com/wavesplatform/dex/grpc/integration/services/WavesBlockchainApiGrpcService.scala

```diff
@@ -33,20 +34,57 @@ import scala.concurrent.Future
 import scala.concurrent.duration.FiniteDuration
 import scala.util.control.NonFatal
 
+// TODO remove balanceChangesBatchLingerMs parameter after release 2.1.2
 class WavesBlockchainApiGrpcService(context: ExtensionContext, balanceChangesBatchLingerMs: FiniteDuration)(implicit sc: Scheduler)
     extends WavesBlockchainApiGrpc.WavesBlockchainApi
     with ScorexLogging {
 
+  private val allSpendableBalances = new ConcurrentHashMap[Address, Map[Asset, Long]]()
+
   // A clean logic requires more actions, see DEX-606
+  // TODO remove after release 2.1.2
   private val balanceChangesSubscribers = ConcurrentHashMap.newKeySet[StreamObserver[BalanceChangesResponse]](2)
+  // TODO rename to balanceChangesSubscribers after release 2.1.2
+  private val realTimeBalanceChangesSubscribers = ConcurrentHashMap.newKeySet[StreamObserver[BalanceChangesFlattenResponse]](2)
+
   private val cleanupTask: Task[Unit] = Task {
     log.info("Closing balance changes stream...")
     // https://github.com/grpc/grpc/blob/master/doc/statuscodes.md
     val shutdownError = new StatusRuntimeException(Status.UNAVAILABLE) // Because it should try to connect to other DEX Extension
-    balanceChangesSubscribers.forEach(_.onError(shutdownError))
+
+    // TODO remove after release 2.1.2
+    balanceChangesSubscribers.forEach { _.onError(shutdownError) }
     balanceChangesSubscribers.clear()
+
+    realTimeBalanceChangesSubscribers.forEach { _.onError(shutdownError) }
+    realTimeBalanceChangesSubscribers.clear()
   }
 
+  private val realTimeBalanceChanges: Coeval[CancelableFuture[Unit]] = Coeval.evalOnce {
+    context.spendableBalanceChanged
+      .map {
+        case (address, asset) =>
+          val newAssetBalance     = context.utx.spendableBalance(address, asset)
+          val maybeAddressBalance = Option(allSpendableBalances get address)
```

Comment:
How about
```scala
Option(allSpendableBalances get address).getOrElse(Map.empty)
```
? This will make the code below easier

Reply:
Fixed